cli/config/configfile/file_test.go

```diff
@@ -380,6 +381,33 @@ func TestGetAllCredentialsCredHelperOverridesDefaultStore(t *testing.T) {
 	assert.Check(t, is.Equal(0, testCredHelper.(*mockNativeStore).GetAllCallCount))
 }
 
+func TestLoadFromReaderWithUsernamePassword(t *testing.T) {
+	configFile := New("test-load")
+	defer os.Remove("test-load")
+
+	want := types.AuthConfig{
+		Username: "user",
+		Password: "pass",
+	}
+	cf := ConfigFile{
+		AuthConfigs: map[string]types.AuthConfig{
+			"example.com/foo": want,
```

Comment:
Can you also test when auth is empty? As this PR fixes that path, it should be tested too, not just the happy path 🤔

Reply:
`auth` *is* empty in this case, which is why we get the Username and Password back (instead of nothing). I've added a case the encodes the `username` and `password` as `auth` to test that it still works.

Reply:
Yep sorry misread it 😅 Thanks for adding the other test case 👍